src/config.rs

```diff
@@ -232,11 +233,23 @@ impl Config {
             .cookie_file
             .unwrap_or_else(|| daemon_dir.join(".cookie"));
 
+        let daemon_auth = match config.auth {
+            Some(user_password) => {
+                let parts: Vec<&str> = user_password.split(":").collect();
+                if parts.len() != 2 {
```

Comment:
I don't believe `:` in password is forbidden. I suggest:

```rust
let colon_pos = user_password.find(':').unwrap_or_else(|| { eprintln!("Error: auth cookie doesn't contain colon"); std::process::exit(1) });
Auth::UserPass(user_password[..colon_pos].to_owned(), user_password[(colon_pos + 1)..].to_owned())
```

Note that my suggestion also intentionally avoids leaking `auth`.

Reply:
You're right - fixed using `str::splitn`.